### PR TITLE
fix(static-site): update static site ts

### PR DESCRIPTION
**Pull request by ritesh**
- Remove the publicly accessible bucket and grant access to cloudfront only

Fixes # 

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by ryparker** [comment]
- Combined changes from https://github.com/aws-samples/aws-cdk-examples/pull/444/files
- Updated README with extra context param `-c accountId=1234567890` to prevent `cdk synth` from throwing `Error: Cannot retrieve value from context provider hosted-zone since account/region are not specified at the stack level.`.
- Updated tsconfig to only include root ts files. This fixes an issue where `tsc` would throw an error when attempting to transpile files generated in `cdk.out/`.
- Upgraded Typescript dependency to `^3.8` to enable usage of `import ... from ...` syntax.
- Replaced deprecated use of `aliasConfiguration` with usage of `loudfront.ViewerCertificate.fromAcmCertificate`
- Updated S3 config to use `blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,` to block public access. This is to promote best practice.

**Comment by ryparker** [comment]
Tested!

**Review by ryparker — APPROVED** [review]
Thanks for contributing 😃

File: typescript/static-site/static-site.ts [hunk 1]
```diff
@@ -49,21 +69,36 @@ export class StaticSite extends Construct {
...
+        // Specifies you want viewers to use HTTPS & TLS v1.1 to request your objects
+        const viewerCertificate = cloudfront.ViewerCertificate.fromAcmCertificate({
+            certificateArn: certificateArn,
+            env: {
+                region: cdk.Aws.REGION,
+                account: cdk.Aws.ACCOUNT_ID
+            },
+            node: this.node,
+            stack: parent
+        },
```
Comment (nerg4l) [inline typescript/static-site/static-site.ts:81]: `ICertificate` requires `metricDaysToExpiry` method but that's not defined on this object. There for ​tsc fails with:

```
cdk/static-site.ts:72:83 - error TS2345: Argument of type '{ certificateArn: string; env: { region: string; account: string; }; node: cdk.ConstructNode; stack: cdk.Stack; }' is not assignable to parameter of type 'ICertificate'.
 ​Property 'metricDaysToExpiry' is missing in type '{ certificateArn: string; env: { region: string; account: string; }; node: cdk.ConstructNode; stack: cdk.Stack; }' but required in type 'ICertificate'.

​72         const viewerCertificate = cloudfront.ViewerCertificate.fromAcmCertificate({
                                                                                     ​~
​73                 certificateArn: certificateArn,
   ​~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
​79                 stack: parent,
   ​~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
​80             },
   ​~~~~~~~~~~~~~

 ​node_modules/@aws-cdk/aws-certificatemanager/lib/certificate.d.ts:28:5
   ​28     metricDaysToExpiry(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
          ​~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   ​'metricDaysToExpiry' is declared here.


Found 1 error.
```

Should I open an issue?

Comment (cinthialandia) [reply typescript/static-site/static-site.ts:81]: I had to add this to my Stack to fix the typescript issue:

```ts
import * as cloudwatch from "@aws-cdk/aws-cloudwatch";

// then on the error:

metricDaysToExpiry: () =>
    new cloudwatch.Metric({
        namespace: "some namespace",
         metricName: "TLS Viewer Certificade Expired",
      }),
```

Comment (ryparker) [reply typescript/static-site/static-site.ts:81]: Thanks for flagging this. I've got a fix here: https://github.com/aws-samples/aws-cdk-examples/pull/479

**Pull request merged**